lib/node_modules/@stdlib/array/base/mskfilter-map/README.md

```diff
@@ -0,0 +1,173 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# mskfilterMap
+
+> Apply a mask to a provided input array and return a new array after applying a mapping function.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var mskfilterMap = require( '@stdlib/array/base/mskfilter-map' );
+```
+
+#### mskfilterMap( x, mask, clbk\[, thisArg] )
+
+Applies a mask to a provided input array and returns a new array after applying a mapping function.
+
+```javascript
+function clbk( value ) {
+    return value * 2;
+}
+
+var x = [ 1, 2, 3, 4 ];
+var m = [ 0, 1, 0, 1 ];
+
+var y = mskfilterMap( x, m, clbk );
+// returns [ 4, 8 ]
+```
+
+The function supports the following parameters:
+
+-   **x**: input array.
+-   **mask**: mask array.
+-   **clbk**: function to apply.
+-   **thisArg**: applied function execution context (_optional_).
+
+To set the applied function's execution context, provide a `thisArg`.
+
+The function **always** returns a new "generic" array.
+
+#### mskfilterMap.assign( x, mask, out, stride, offset, clbk\[, thisArg] )
+
+Applies a mask and mapping function to a provided input array and assigns results to elements in a provided output array.
+
+```javascript
+function clbk( value ) {
+    return value * 2;
+}
+
+var x = [ 1, 2, 3, 4 ];
+var m = [ 0, 1, 0, 1 ];
+
+var out = [ 0, 0, 0, 0 ];
+var arr = mskfilterMap.assign( x, m, out, -2, out.length-1, clbk );
+// returns [ 0, 8, 0, 4 ]
+
+var bool = ( arr === out );
+// returns true
+```
+
+The function supports the following parameters:
+
+-   **x**: input array.
+-   **mask**: mask array.
+-   **out**: output array.
+-   **stride**: output array stride.
+-   **offset**: output array offset.
+-   **clbk**: function to apply.
+-   **thisArg**: applied function execution context (_optional_).
+
+To set the applied function's execution context, provide a `thisArg`.
```

Comment:
You can remove this line once you've moved the example below up to the previous API.